internal/metrics/alerts/collector.go

```diff
@@ -0,0 +1,75 @@
+package alerts
+
+const CollectorPrometheusAlert = `
+"groups":
+- "name": "logging_collector.alerts"
+  "rules":
+  - "alert": CollectorNodeDown
+    "annotations":
+      "message": "Prometheus could not scrape {{ $labels.container }}-{{ $labels.app_kubernetes_io_name }} for more than 10m."
+      "summary": "Collector cannot be scraped"
+    "expr": |
+      up{app_kubernetes_io_component = "collector", container = "collector"} == 0 or absent(up{app_kubernetes_io_component="collector", container="collector"}) == 1
+    "for": "10m"
+    "labels":
+      "service": "collector"
+      "severity": "critical"
+      "namespace": "openshift-logging"
+  - "alert": "CollectorHighErrorRate"
+    "annotations":
+      "message": |-
+        {{ $value }}% of records have resulted in an error by {{ $labels.app_kubernetes_io_name }} collector component.
+      "summary": "{{ $labels.app_kubernetes_io_name }} component errors are very high"
+    "expr": |
+      100 * (
+          collector:num_errors:sum_rate
+        /
+          collector:received_events:sum_rate
+        ) > 10
+    "for": 15m
+    "labels":
+      "service": "collector"
+      "severity": "critical"
+      "namespace": "openshift-logging"
+  - "alert": "CollectorVeryHighErrorRate"
+    "annotations":
+      "message": |-
+        {{ $value }}% of records have resulted in an error by {{ $labels.app_kubernetes_io_name }} collector component.
+      "summary": "{{ $labels.app_kubernetes_io_name }} collector component errors are very high"
+    "expr": |
+      100 * (
+          collector:num_errors:sum_rate
+        /
+          collector:received_events:sum_rate
+        ) > 25
+    "for": 15m
+    "labels":
+      "service": "collector"
+      "severity": "critical"
+      "namespace": "openshift-logging"
+  - "alert": "FluentdQueueLengthIncreasing"
+    "annotations":
+      "message": "For the last hour, fluentd {{ $labels.pod }} output '{{ $labels.plugin_id }}' average buffer queue length has increased continuously."
+      "summary": "Fluentd pod {{ $labels.pod }} is unable to keep up with traffic over time for forwarder output {{ $labels.plugin_id }}."
+    "expr": |
+      sum by (pod,plugin_id) ( 0 * (deriv(fluentd_output_status_emit_records[1m] offset 1h)))  + on(pod,plugin_id)  ( deriv(fluentd_output_status_buffer_queue_length[10m]) > 0 and delta(fluentd_output_status_buffer_queue_length[1h]) > 1 )
+    "for": "1h"
+    "labels":
+      "service": "collector"
+      "severity": "Warning"
+      namespace: "openshift-logging"
+- "name": "logging_clusterlogging_telemetry.rules"
+  "rules":
+  - "expr": |
+      sum by(cluster)(log_collected_bytes_total)
+    "record": "cluster:log_collected_bytes_total:sum"
```

Comment:
@xperimental what is the convention for namespacing here?  We should maybe consider how to make this "Red Hat logging specifc"